src/expr/src/interpret.rs

```diff
@@ -1550,6 +1576,46 @@ mod tests {
         assert!(range_out.may_contain(Datum::Null));
     }
 
+    #[mz_ore::test]
+    fn test_inequality() {
+        let arena = RowArena::new();
+
+        let expr = MirScalarExpr::CallBinary {
+            func: BinaryFunc::Gte,
+            expr1: Box::new(MirScalarExpr::Column(0)),
+            expr2: Box::new(MirScalarExpr::CallUnmaterializable(
+                UnmaterializableFunc::MzNow,
+            )),
+        };
+
+        let relation = RelationType::new(vec![ScalarType::MzTimestamp.nullable(true)]);
+        let mut interpreter = ColumnSpecs::new(&relation, &arena);
+        interpreter.push_column(
+            0,
+            ResultSpec::value_between(
+                Datum::MzTimestamp(1704736444949u64.into()),
+                Datum::MzTimestamp(1704736444949u64.into()),
+            )
+            .union(ResultSpec::null()),
+        );
+        interpreter.push_unmaterializable(
+            UnmaterializableFunc::MzNow,
+            ResultSpec::value_between(
+                Datum::MzTimestamp(1704738791000u64.into()),
+                Datum::MzTimestamp(18446744073709551615u64.into()),
+            ),
+        );
+
+        let range_out = interpreter.expr(&expr).range;
+        assert!(
+            !range_out.fallible,
+            "<= function should not error on non-error input"
+        );
+        assert!(!range_out.may_contain(Datum::True));
+        assert!(range_out.may_contain(Datum::False));
+        assert!(range_out.may_contain(Datum::Null));
```

Comment:
Can you explain  again how this test is hitting the new `flat_map` case? As far as I can see you are interpreting an expression 

```
#0 >= mz_now()
```
under the assumption that:

```
#0 ∈ [1704736444949u64, 1704736444949u64]
mz_now() ∈ [1704738791000u64, 18446744073709551615u64] 
```
which should always evaluate to `Datum::False`.

Reply:
Sure! The important bit is that #0 can also be null. During the outermost flat_map we'll consider the `null < [1704738791000u64, 18446744073709551615u64]` case, which we previously didn't handle well  but now correctly evaluates to `null`. (And the non-null case always evaluates to `False`, as you say.)